registration-processor/core-processor/registration-processor-manual-verification-stage/src/main/java/io/mosip/registration/processor/manual/verification/service/impl/ManualVerificationServiceImpl.java

```diff
@@ -543,6 +555,120 @@ public String getFilename() {
+					if (policyMap.containsValue(docName)) {
+						for(Documents docs:documents) {
+							if(docs.getCategory().equalsIgnoreCase(docName)) {
+								if (requestDto.getDocuments() != null)
+									requestDto.getDocuments().put(docmap.get(MappingJsonConstants.VALUE).toString(), docs.getValue());
+								else {
+									Map<String, String> docMap = new HashMap<>();
+									docMap.put(docmap.get(MappingJsonConstants.VALUE).toString(), docs.getValue());
+									requestDto.setDocuments(docMap);
+								}
+							}
+						}			
+					}
+				}
+			}
+			
+			JSONObject regProcessorIdentityJson = utility.getRegistrationProcessorMappingJson(MappingJsonConstants.IDENTITY);
+			String individualBiometricsLabel = JsonUtil.getJSONValue(
+					JsonUtil.getJSONObject(regProcessorIdentityJson, MappingJsonConstants.INDIVIDUAL_BIOMETRICS),
+					MappingJsonConstants.VALUE);
+
+			if (policyMap.containsValue(individualBiometricsLabel)) {
+				for(Documents docs:documents) {
+				if(docs.getCategory().equalsIgnoreCase(individualBiometricsLabel)) {
+				requestDto.setBiometrics(docs.getValue() != null ? docs.getValue() : null);
+			}
+			}
+			}
+			JSONObject auditsJson = utility.getRegistrationProcessorMappingJson(MappingJsonConstants.AUDITS);
+			String auditsLabel = JsonUtil.getJSONValue(auditsJson, MappingJsonConstants.VALUE);
+			if (policyMap.containsValue(auditsLabel)) {
+				for(Documents docs:documents) {
+				if(docs.getCategory().equalsIgnoreCase(auditsLabel)) {
+				requestDto.setAudits(docs.getValue() != null ? docs.getValue() : null);
+			}
+			}
+			}
+			JSONObject metaInfoJson = utility.getRegistrationProcessorMappingJson(MappingJsonConstants.METAINFO);
+			String metaInfoLabel = JsonUtil.getJSONValue(auditsJson, MappingJsonConstants.VALUE);
+			if (policyMap.containsValue(metaInfoLabel)) {
+				for(Documents docs:documents) {
+				if(docs.getCategory().equalsIgnoreCase(metaInfoLabel)) {
+				requestDto.setMetaInfo(docs.getValue() != null ? docs.getValue() : null);
+			}
+			}
+			}
+		
+		}
+
+		String req = JsonUtils.javaObjectToJsonString(requestDto);
+
+		MultiValueMap<String, Object> map = new LinkedMultiValueMap<>();
+		map.add("name", MANUAL_VERIFICATION);
+		map.add("filename", MANUAL_VERIFICATION);
+
+		ByteArrayResource contentsAsResource = new ByteArrayResource(req.getBytes()) {
+			@Override
+			public String getFilename() {
+				return MANUAL_VERIFICATION;
+			}
+		};
+		map.add("file", contentsAsResource);
+
+		List<String> pathSegments = new ArrayList<>();
+		pathSegments.add(policyId);
+		pathSegments.add(subscriberId);
+		URL dataShareUrl = null;
+		String protocol = PROTOCOL;
+		String url = null;
+
+		if (httpProtocol != null && !httpProtocol.isEmpty()) {
+			protocol = httpProtocol;
+		}
+
+		dataShareUrl = new URL(protocol, internalDomainName, env.getProperty(ApiName.DATASHARECREATEURL.name()));
```

Comment:
Here datashare url creation code was already  there. You have created lot of duplicated code. Can you please use existing code to create datashare. You can create separate methods for -
getDocuments()
getDemographic()
getAudits()
getDatashare()

like this and reuse methods in both idrepo and packetmanager cases

Reply:
I have created separate methods for getDocuments() and getDemographic() and  getDatashare().